### PR TITLE
[Tests] Add set_env test to kubejob runtime

### PR DESCRIPTION
**Pull request by Yacouby**

### CODE REVIEW

File: tests/api/runtimes/base.py [hunk 1]
```diff
@@ -157,10 +158,16 @@ def _assert_function_config(
     @staticmethod
     def _assert_pod_env(pod_env, expected_variables):
         for env_variable in pod_env:
-            name = env_variable["name"]
+            if isinstance(env_variable, V1EnvVar):
```
Comment (theSaarco) [inline tests/api/runtimes/base.py:161]: You're doing the `isinstance` check in multiple places. It's better to convert `env_variable` in a single place:
```
if isinstance(env_variable, dict):
    env_variable = V1EnvVar(name=env_variable["name"], value=env_variable["value"])
```
And then you can just use the `V1EnvVar` accessors.

File: tests/api/runtimes/test_kubejob.py [hunk 2]
```diff
@@ -178,3 +178,12 @@ def my_func(context):
 
         self._execute_run(runtime)
         self._assert_pod_create_called(expected_code=expected_code)
+
+    def test_set_env(self, db: Session, client: TestClient):
+        runtime = self._generate_runtime()
+        env = {"MLRUN_LOG_LEVEL": "DEBUG", "IMAGE_HEIGHT": "128"}
+        runtime.set_env("MLRUN_LOG_LEVEL", "DEBUG")
+        runtime.set_env("IMAGE_HEIGHT", "128")
+
```
Comment (theSaarco) [inline tests/api/runtimes/test_kubejob.py:187]: Better to define `env` as you did and then reference it. This way if you want to change something in its values you don't need to do it in two places. Replace the `set_env` lines with something like:
```
for env_variable in env:
    runtime.set_env(env_variable, env[env_variable])
```

File: tests/api/runtimes/base.py [hunk 3]
```diff
@@ -157,10 +158,16 @@ def _assert_function_config(
...
+                name = env_variable.name
+            else:
+                name = env_variable["name"]
             if name in expected_variables:
                 if expected_variables[name]:
-                    assert expected_variables[name] == env_variable["value"]
+                    if isinstance(env_variable, V1EnvVar):
+                        assert expected_variables[name] == env_variable.value
+                    else:
+                        assert expected_variables[name] == env_variable["value"]
```
Comment (theSaarco) [inline tests/api/runtimes/base.py:170]: See comment above.

File: tests/api/runtimes/base.py [hunk 4]
```diff
@@ -301,7 +308,10 @@ def _assert_pod_create_called(
         expected_env["MLRUN_NAMESPACE"] = self.namespace
         self._assert_pod_env(pod_env, expected_env)
         for env_variable in pod_env:
-            if env_variable["name"] == "MLRUN_EXEC_CONFIG":
+            if (
+                not isinstance(env_variable, V1EnvVar)
```
Comment (theSaarco) [inline tests/api/runtimes/base.py:312]: Why are you limiting yourself like this? Again here I would use the method mentioned above and convert. This way it doesn't matter what type is within the structure.

**Review by theSaarco — CHANGES_REQUESTED** [review]
Looks good. Would do some changes mostly to make the code simpler and more readable.

**Review by theSaarco — APPROVED** [review]

**Review by Hedingber — CHANGES_REQUESTED** [review]
@Yacouby 
Looks awesome!
Please add a test for the case where you call `set_env` with one value for a certain key
and then calling it again with the same key but a different value (assert the second value is used)

**Review by Hedingber — APPROVED** [review]
@Yacouby 
Looks awesome!
Please add a test for the case where you call `set_env` with one value for a certain key
and then calling it again with the same key but a different value (assert the second value is used)

**Pull request merged**